shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -527,7 +555,45 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
           val iter = unsafeRows.iterator
           val unsafeProjection =
             GenerateUnsafeProjection.generate(requestedSchema, cacheSchema)
-          iter.map(unsafeProjection)
+          if (hasUnsupportedType) {
+            new Iterator[InternalRow]() {
+              val wrappedIter = iter
+              val newRow = new GenericInternalRow(cacheSchema.length)
+
+              override def hasNext: Boolean = wrappedIter.hasNext
+
+              override def next(): InternalRow = {
+                //read a row and convert it to what the caller is expecting
+                val row = wrappedIter.next()
+                var newIndex = 0
+                origCacheSchema.indices.map { index =>
+                  val attribute = origCacheSchema(index)
+                  attribute.dataType match {
+                    case CalendarIntervalType => {
+                      // create a CalendarInterval based on the next three values
+                      val interval = new CalendarInterval(row.getInt(newIndex),
+                      row.getInt(newIndex + 1), row.getLong(newIndex + 2))
+                      newIndex += 3
+                      newRow.setInterval(index, interval)
+                    }
+                    case NullType => {
+                      newRow.setNullAt(index)
+                      newIndex += 1
+                    }
+                    case _ => {
+                      newRow.update(index, row.get(newIndex, attribute.dataType))
```

Comment:
What about an array of `CalendarIntervalType` or a Struct that includes one of them? This does not appear to deal with unsupported types inside of nested objects.

Reply:
It will be great if we can get this in before we tackle the nested types. Let me know what you think

Reply:
That is fine to have nested types be a follow on issue.

Reply:
#1092